test/unit/org/apache/cassandra/io/util/Files.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.io.util;
+
+import java.io.IOException;
+import java.nio.file.FileSystem;
+
+import com.google.common.base.StandardSystemProperty;
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+
+import org.apache.cassandra.io.filesystem.ForwardingFileSystem;
+import org.apache.cassandra.io.filesystem.ListenableFileSystem;
+
+public class Files
```

Comment:
Just thinking about the general usage of this class...

1.) I'd consider naming it `Filesystem` or `Filesystems`

2.) Would it be better to separate filesystem creation and replacement? For example...

```
FileSystem jim = Filesystems.newJimFs();
Filesystems.replace(jim, true); // where in replace(FileSystem fs, boolean restoreOnClose), restoreOnClose specifies whether to restore the original FS
```

Reply:
`Filesystems.replace` is a duplicate of `org.apache.cassandra.io.util.File#unsafeSetFilesystem`... and `global` method kinda does what you are asking, but when `true`...

so in short, if you want `false` just call `org.apache.cassandra.io.util.File#unsafeSetFilesystem`.  If you want `true` call `global`?

Reply:
Right, you can get all this behavior w/ the existing methods...just trying to put an alternate API on top of it.

Reply:
Either way, the class name `Files` seems suboptimal.

Reply:
> Right, you can get all this behavior w/ the existing methods...just trying to put an alternate API on top of it.

in this case I feel 2 methods are better than a boolean; easier to document and I feel easier to read while reviewing code